plugin/encode.go

```diff
@@ -360,11 +362,14 @@ func (s *Server) encodeVariable(variable *tfconfigs.Variable) *tfplugin.Variable
 		validations[i] = s.encodeVariableValidation(v)
 	}
 
+	defaultVal, _ := msgpack.Marshal(variable.Default, variable.Default.Type())
+	// We need to use json because cty.DynamicPseudoType cannot be encoded by gob
+	typeVal, _ := json.MarshalType(variable.Type)
```

Comment:
We may need to extend `encodeVariable` to properly return this error to the caller.